src/test/java/co/rsk/federate/signing/hsm/client/PowHSMSigningClientBtcTest.java

```diff
@@ -423,36 +423,44 @@ private void addCommonPegoutLogs(List<LogInfo> logs, BtcTransaction pegoutBtcTx)
 
     private BtcTransaction createPegout(List<Coin> outpointValues,
         List<Address> destinationAddresses, boolean segwit) {
-        BtcTransaction fundingTransaction = new BtcTransaction(btcMainnetParams);
-        fundingTransaction.addInput(BitcoinTestUtils.createHash(1), FIRST_OUTPUT_INDEX,
-            new Script(new byte[]{}));
 
-        for (Coin outpointValue : outpointValues) {
-            fundingTransaction.addOutput(outpointValue, oldFederation.getAddress());
-        }
-
-        BtcTransaction pegoutBtcTx = new BtcTransaction(btcMainnetParams);
+        BtcTransactionBuilder btcTransactionBuilder = new BtcTransactionBuilder();
 
+        // TODO: improve this method to create a more realistic btc segwit transaction
+        //  once {@link SignerMessageBuilder#getSigHashByInputIndex(int)} is refactored to support segwit
         Script inputScriptThatSpendsFromTheFederation = createBaseInputScriptThatSpendsFromTheFederation(
             oldFederation);
 
         Coin fee = Coin.MILLICOIN;
-        for (int outputIndex = 0; outputIndex < outpointValues.size(); outputIndex++) {
-            TransactionInput addedInput = pegoutBtcTx.addInput(
-                fundingTransaction.getOutput(outputIndex));
-
-            addedInput.setScriptSig(inputScriptThatSpendsFromTheFederation);
-
+        for (int inputIndex = 0; inputIndex < outpointValues.size(); inputIndex++) {
+            Coin outpointValue = outpointValues.get(inputIndex);
+            Coin amountToSend = outpointValue.minus(fee);
+            // Iterate over the addresses using inputIndex % addresses.size() to have outputs to different addresses
+            Address destinationAddress = destinationAddresses.get(
+                inputIndex % destinationAddresses.size());
+
+            TransactionInput txInput = btcTransactionBuilder.createInputBuilder()
+                .withAmount(outpointValue).withOutpointIndex(inputIndex)
+                .withScriptSig(inputScriptThatSpendsFromTheFederation)
+                .build();
+
+            btcTransactionBuilder
+                .withInput(
+                    txInput
+                )
+                .withOutput(amountToSend, destinationAddress);
+
+            // TODO: change this dummy witness for a real witness once segwit is fully implemented in bitcoinj-thin
+            // make it a segwit tx by adding a single witness
             if (segwit) {
-                addWitness(pegoutBtcTx, outputIndex);
-            }
+                TransactionWitness witness = new TransactionWitness(1);
+                witness.setPush(0, new byte[]{1});
```

Comment:
```suggestion
                witness.setPush(ScriptOpCodes.OP_0, new byte[]{1});
```

No big deal, but maybe this provides a little more context.

Reply:
This does not pass a `ScriptOpCode`. It passes the index of input to which the witness belongs.

Reply:
Gotcha.